pydicom/values.py

```diff
@@ -377,7 +378,12 @@ def convert_numbers(
     length = len(byte_string)
 
     if length % bytes_per_value != 0:
-        logger.warning("Expected length to be even multiple of number size")
+        raise BytesLengthException("Expected total bytes to be an even "
+                                   f"multiple of bytes per value. Instead "
+                                   f"received {byte_string} with length "
+                                   f"{length} and struct format "
+                                   f"'{struct_format}' which corresponds to "
+                                   f"bytes per value of {bytes_per_value}.")
 
```

Comment:
Just for style again, the open bracket should be last character on the line, the block indented only one level, and the closing bracket same indent as 'raise'